File: response-time/controllers/go/controller.go
package main

import (
	"log"
	"net"
)

func main() {
	laddr, err := net.ResolveUDPAddr("udp", ":8080")
	if err != nil {
		log.Fatalf("Unable to resolve local UDP address: %s\n", err.Error())
	}

	conn, err := net.ListenUDP("udp", laddr)
	if err != nil {
		log.Fatalf("Unable to listen for packets: %s\n", err.Error())
	}
	defer conn.Close()

	for {
		buf := make([]byte, 1024)
		_, raddr, err := conn.ReadFromUDP(buf)
		if err != nil {
			log.Printf("Unable to receive packet: %s\n", err.Error())
		}
		log.Println("Packet received")
		conn.WriteToUDP([]byte("pong"), raddr)
	}
}

File: response-time/controllers/rs/controller.rs
use std::net;

fn main() {
    let socket = net::UdpSocket::bind("0.0.0.0:8080").expect("Unable to bind UDP socket");
    
    let mut buf = [0; 1024];
    loop {
        match socket.recv_from(&mut buf) {
            Ok((_, src)) => {
                println!("Packet received");
                socket.send_to(b"pong", src).expect("Unable to send packet");
            },
            Err(e) => {
                eprintln!("Unable to receive packet: {}", e);
            }
        }
    }
}
